cmake/Modules/CheckTypeAlignment.c.in

```diff
@@ -0,0 +1,56 @@
+@headers@
```

Comment:
This is horrible, and looking ahead a little bit (#1080) it is really only used for determining the alignment of an eventset. The only values supported are 1,2,4,8. It is almost a given that at least one bit will be an integer. So we're literally talking possibly wasting 4 bytes if the union just uses an `uint64_t` for alignment.

In reality, that won't suffice for the machine that @dodsonmg provided some fixes for, where a pointer is larger (see e.g., 004932817e111597db65e6b332f92ae9bbe9fe30). The good thing of the mechanism in this PR is that won't fail silently but simply fail to compile. Still, if the union used to hide implementation details were just to use both `uint64_t` and `void *` to ensure alignment requirements are met, we should be ok.

So we can (and should) avoid all this ...

(And when it really becomes more difficult, there are also the gcc/clang/visual studio attributes/declspecs for tweaking alignment.)

Reply:
You mean drop the check alignment stuff? Yes, it's currently only used for the eventset, but the intent is to hide implementation details in more places eventually. i.e. not leaking system headers. The reason I'm doing it here is because exporting `wepoll` identifiers seemed like an even worse idea than leaking system headers. At some point, that means we have to check the alignment of a type somewhere and it cannot be done at compile time of ddsc. I could implement something to compare against various other types (including `void *`) and export the correct type directly from the CMake module.

But what is it that you want to avoid? It's either leaking headers or checking alignment right?

Reply:
The alignment checks have been removed from this PR all together as they're not needed here. I'll do it in a more pragmatic way in PR #1080.